core/src/main/scala/caliban/parsing/adt/Selection.scala

```diff
@@ -1,13 +1,15 @@
 package caliban.parsing.adt
 
 import caliban.InputValue
+import caliban.Scala3Annotations.threadUnsafe
 import caliban.parsing.adt.Type.NamedType
 
 import scala.util.hashing.MurmurHash3
 
-sealed trait Selection extends Serializable {
-  // TODO: Kept for binary compatibility
-  @transient override lazy val hashCode: Int = super.hashCode()
+sealed trait Selection extends Product with Serializable {
+
+  @transient @threadUnsafe
+  override final lazy val hashCode: Int = MurmurHash3.productHash(this)
```

Comment:
Also made `Selection` extend Product here since this targets the `series/3.x` branch